content/en/docs/architecture/timeouts.md

```diff
@@ -1,24 +1,55 @@
+description: An overview of job execution timeouts, how to configure them and what the test workload is expected to do when handling them. 
 ---
 
-A typical OpenShift CI job runs [ci-operator](/docs/architecture/ci-operator/) in a pod which for each [step](/docs/architecture/step-registry/#step) of the job creates a pod, executing the commands defined by `tests.steps.test.commands` of a step in the job. A _test process_ is a process which runs such commands. There are timeouts in various phases of a job execution. In this document, we will explain those timeouts, show the logs if they occur in a job, and how the test process will be notified, and how to modify the values of those timeouts.
+# Overview
 
-## Job Timeout
+Job execution timeouts and preemptions occur regularly in the OpenShift CI infrastructure, and a well-formed job must be able to handle these events. The most common tasks that jobs should execute when reacting to interruptions are to generate output artifacts and logs to aid developers investigating the job failure in the future and to clean up any shared infrastructure or cloud resources that the job may be using.
 
-The job timeout can be considered as the overall time for `ci-operator` to run. When a job fails due to the job timeout, the job log contains:
+# Architecture
 
+## How Interruptions (Preemptions and Timeouts) Occur
+
+Fully understanding the set of timeout conditions requires an overview of how jobs are executed. When a job is triggered by Prow, our top-level job orchestrator, a `Pod` is launched on the central CI build farm clusters to execute it. In OpenShift CI, this `Pod` runs [`ci-operator`](/docs/architecture/ci-operator), our job-specific orchestrator, with a cursory set of parameters (please execute job _foo_ for the _main_ branch of a specific repository). `ci-operator` in turn creates a `Namespace` on the build farm cluster in which more `Pods` execute individual [steps](/docs/architecture/step-registry/#step). Each `Pod` executes the commands as configured in `tests.steps.test.commands`, these commands are known as the _test process_. As a job author, you control the content of these commands, and the set of steps which run in the job.
+
+Timeouts occur for a job when the overall job or some subset of the job exceeds the configured budget for execution. Preemptions occur when a new version of the job is triggered, and the previous version is rendered obsolete, as happens when new code is pushed to a Pull Request and previous commits are no longer important to test. Timeouts and preemptions are implemented via `Pod` deletions on our build farm clusters. Therefore, all interruption events will obey the basic rules of Kubernetes [graceful termination](https://cloud.google.com/blog/products/containers-kubernetes/kubernetes-best-practices-terminating-with-grace): every process is given a `SIGTERM`, some time to react to it gracefully, then given a `SIGKILL`.
```

Comment:
It might be worth tightening up "every process".  Per [this blog][1], it is "the main process (PID 1) in each container".  I expect "main process in each container" is the main point, because in pods which opt to skip PID namespacing, those processes would have higher PIDs.  But for the purpose of this doc, whatever we want to say so that folks think through distributing SIGTERM down to the test process's children and ancestors.

[1]: https://pracucci.com/graceful-shutdown-of-kubernetes-pods.html